src/Elf.h

```diff
@@ -0,0 +1,395 @@
+#ifndef HALIDE_ELF_H
+#define HALIDE_ELF_H
+
+#include <memory>
+#include <vector>
+#include <list>
+#include <string>
+#include <iterator>
+#include <limits>
+
+namespace Halide {
+namespace Internal {
+namespace Elf {
+
+// This ELF parser mostly deserializes the object into a graph
```

Comment:
If you have an ELF spec or something that documents the meaning of things like STB_GLOBAL, please add a link to it here.